### PR TITLE
feat: check fast-track approvals

### PR DESCRIPTION
**Pull request by Mesteery**

### CODE REVIEW

File: lib/pr_checker.js [hunk 1]
```diff
@@ -168,6 +168,25 @@ class PRChecker {
...
+        return false;
+      }
+      const [, requestor] = comment.bodyText.match(/^Fast-track has been requested by @(.+?)\. Please 👍 to approve\.$/);
+      const collaborators = [...this.data.collaborators.values()]
+        .map((c) => c.login);
+      if (!comment.reactions.nodes.some((r) =>
+        r.user.login !== requestor &&
+        collaborators.includes(r.user.login))) {
+        cli.error('The fast-track request requires' +
+                  " at least one collaborator's 👍.");
```
Comment (Trott) [inline lib/pr_checker.js:185]: Fast tracking requires two collaborator approvals.

> The pull request can be fast-tracked if two collaborators approve the fast-tracking request. To land, the pull request itself still needs two collaborator approvals and a passing CI.



Ref: https://github.com/nodejs/node/blob/86099a375af95f519ae699b68ce101aa20236c5f/doc/guides/collaborator-guide.md#waiting-for-approvals

Would it be possible to implement logic such that it checks to see if the person that requested fast-tracking is also the author of the PR and, if so, checks for two collaborator fast-track 👍 reactions, and otherwise accepts one collaborator fast-track 👍 reaction as long as it's not a 👍 from either the PR author or the collaborator proposing fast-tracking?

In other words:

```
If fast-track requester is PR author:
    Require 2 👍 reactions from collaborators who are not the PR author
else:
    Require 1 👍 reaction from a collaborator who is not the PR author or the collaborator who requested fast-tracking
```

**Comment by Trott** [comment]
Thanks for tackling this problem. The absence of fast-track confirmation is definitely a good loophole to fix!

**Review by targos — APPROVED** [review]

File: lib/pr_checker.js [hunk 2]
```diff
@@ -9,6 +9,8 @@ const WAIT_TIME_SINGLE_APPROVAL = 24 * 7;
 
 const GITHUB_SUCCESS_CONCLUSIONS = ['SUCCESS', 'NEUTRAL', 'SKIPPED'];
 
+const FAST_TRACK_RE = /^Fast-track has been requested by @(.+?)\. Please 👍 to approve\.$/;
```
Comment (aduh95) [inline lib/pr_checker.js:12]: Would it be possible to loosen this regex so if the wording of the comment changes in the future, it doesn't require to update ncu?

File: lib/pr_checker.js [hunk 3]
```diff
@@ -168,6 +170,32 @@ class PRChecker {
...
+    if (isFastTracked) {
+      const comment = this.comments.find((c) =>
+        FAST_TRACK_RE.test(c.bodyText));
+      if (!comment) {
+        cli.error('Unable to find the fast-track request comment.');
+        return false;
+      }
+      const [, requester] = comment.bodyText.match(FAST_TRACK_RE);
+      const collaborators = [...this.data.collaborators.values()]
+        .map((c) => c.login);
```
Comment (aduh95) [inline lib/pr_checker.js:182]: nit: we can skip the intermediary array creation.
```suggestion
      const collaborators = Array.from(this.data.collaborators.values(),
                                       (c) => c.login);
```

File: lib/pr_checker.js (hunk 2 above)
Comment (Mesteery) [reply lib/pr_checker.js:12]: I thought about it, but I'm afraid it would match a comment that is not a fast-track request (eg. if someone quotes the fast-track request comment).

**Comment by Mesteery** [comment]
ping @Trott

File: lib/pr_checker.js [hunk 4]
```diff
@@ -168,6 +170,32 @@ class PRChecker {
       }
     }
 
+    if (isFastTracked) {
+      const comment = this.comments.find((c) =>
+        FAST_TRACK_RE.test(c.bodyText));
+      if (!comment) {
+        cli.error('Unable to find the fast-track request comment.');
+        return false;
+      }
```
Comment (Trott) [inline lib/pr_checker.js:179]: This is probably OK but just in case: There is a window, often of several minutes, where an issue is labeled with `fast-track` but the comment has not yet appeared. So this error may happen legitimately. Maybe that's fine? Or maybe we just need to mention it in the error message (perhaps by adding "Please try again in a few minutes." or something like that?).

**Review by Trott — APPROVED** [review]

File: lib/pr_checker.js (hunk 4 above)
Comment (Mesteery) [reply lib/pr_checker.js:179]: If there is no comment it means that there is no approval and therefore the PR can not be landed. So I think it's useless.
